File: spring-boot-3tier-form/src/main/java/com/adjh/springboot3tierform/model/dto/UserDto.java
package com.adjh.springboot3tierform.model.dto;

import lombok.*;

/**
 * 사용자 DTO
 *
 * @author : jonghoon
 * @fileName : UserDto
 * @since : 5/15/24
 */
@Getter
@ToString
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class UserDto {

    private Long userSq;
    private String userId;
    private String userNm;
    private String userSt;


    @Builder(toBuilder = true)
    public UserDto(Long userSq, String userId, String userNm, String userSt) {
        this.userSq = userSq;
        this.userId = userId;
        this.userNm = userNm;
        this.userSt = userSt;
    }
}

File: kotlin-spring-boot-form/src/main/kotlin/com/blog/kotlinspringbootform/model/dto/UserDto.kt
package com.blog.kotlinspringbootform.model.dto

/**
 * tb_user 테이블과 매핑되는 DTO 클래스입니다.
 *
 * @author : jonghoon
 * @fileName : UserDto
 * @since : 10/1/24
 */
data class UserDto(
    val userSq: Int,    // 사용자 시퀀스
    val userId: String, // 사용자 아이디
    val userPw: String, // 사용자 패스워드
    val userNm: String, // 사용자 이름
    val userSt: String, // 사용자 상태
)
